aea/cli/packages.py

```diff
@@ -0,0 +1,374 @@
+
+"""Package manager."""
+
+
+import json
+import logging
+import shutil
+import sys
+import traceback
+from collections import OrderedDict
+from pathlib import Path
+from typing import Callable, Optional
+from typing import OrderedDict as OrderedDictType
+from typing import cast
+
+import click
+
+from aea.cli.utils.context import Context
+from aea.cli.utils.decorators import pass_ctx
+from aea.configurations.base import PackageConfiguration
+from aea.configurations.data_types import PackageId, PackageType
+from aea.configurations.loader import load_configuration_object
+from aea.helpers.dependency_tree import DependencyTree
+from aea.helpers.fingerprint import check_fingerprint
+from aea.helpers.io import open_file
+from aea.helpers.ipfs.base import IPFSHashOnly
+
+
+try:
+    from aea_cli_ipfs.registry import fetch_ipfs  # type: ignore
+
+    IS_IPFS_PLUGIN_INSTALLED = True
+except (ImportError, ModuleNotFoundError):
+    IS_IPFS_PLUGIN_INSTALLED = False
+
+PACKAGES_FILE = "packages.json"
+
+
+@click.group("packages")
+@click.pass_context
+def package_manager(
+    click_context: click.Context,  # pylint: disable=unused-argument
+) -> None:
+    """Local package manager."""
+
+
+@package_manager.command()
+@pass_ctx
+@click.option(
+    "--update-packages",
+    is_flag=True,
+    help="Update packages if the calculated hash for a package does not match the one in the packages.json.",
+)
+@click.option(
+    "--update-hashes",
+    is_flag=True,
+    help="Update hashes in the packages.json if the calculated hash for a package does not match the one in the packages.json",
+)
+def sync(ctx: Context, update_packages: bool, update_hashes: bool) -> None:
+    """Sync local packages."""
+
+    if not IS_IPFS_PLUGIN_INSTALLED:
+        raise click.ClickException(
+            "Please install ipfs plugin using `pip3 install open-aea-cli-ipfs`"
+        )
+
+    if update_hashes and update_packages:
+        raise ValueError(
+            "You cannot use both `--update-hashes` and `--update-packages` at the same time."
+        )
+
+    packages_dir = Path(ctx.registry_path)
+    PackageManager.from_dir(packages_dir).sync(
+        update_packages=update_packages,
+        update_hashes=update_hashes,
+    ).update_package_hashes().dump()
+
+
+@package_manager.command(name="lock")
+@click.option(
+    "--check",
+    is_flag=True,
+    help="Check packages.json",
+)
+@pass_ctx
+def lock_packages(ctx: Context, check: bool) -> None:
+    """Lock packages"""
+
+    packages_dir = Path(ctx.registry_path)
+
+    if check:
+        packages_dir = Path(ctx.registry_path)
+        click.echo("Verifying packages.json")
+        return_code = PackageManager.from_dir(packages_dir).verify()
+
+        if return_code:
+            click.echo("Verification failed.")
+        else:
+            click.echo("Verification successful")
+
+        sys.exit(return_code)
+
+    PackageManager(packages_dir).update_package_hashes().dump()
+
+
+def load_configuration(
```

Comment:
We probably have this functionality already in the configurations module. If not, move it there